testing/e2e/e2e_test.go

```diff
@@ -153,3 +187,272 @@ func (suite *BaseE2ETestSuite) FleetServerStatusOK(ctx context.Context, url stri
+	suite.Require().NoError(err)
+	req.SetBasicAuth(suite.elasticUser, suite.elasticPass)
+	req.Header.Set("kbn-xsrf", "e2e-setup")
+	resp, err := suite.client.Do(req)
+	suite.Require().NoError(err)
+	suite.Require().Equal(http.StatusOK, resp.StatusCode)
+
+	var obj struct {
+		ActionID string `json:"actionId"`
+	}
+	err = json.NewDecoder(resp.Body).Decode(&obj)
+	suite.Require().NoError(err)
+	resp.Body.Close()
+	suite.Require().NotEmpty(obj.ActionID)
+	return obj.ActionID
+}
+
+func (suite *BaseE2ETestSuite) VerifyAgentInKibana(ctx context.Context, id string) {
+	req, err := http.NewRequestWithContext(ctx, "GET", "http://localhost:5601/api/fleet/agents/"+id, nil)
+	suite.Require().NoError(err)
+	req.SetBasicAuth(suite.elasticUser, suite.elasticPass)
+	req.Header.Set("kbn-xsrf", "e2e-setup")
+	resp, err := suite.client.Do(req)
+	suite.Require().NoError(err)
+	resp.Body.Close()
+	suite.Require().Equal(http.StatusOK, resp.StatusCode, "expected to find agent in fleet api")
+}
+
+func (suite *BaseE2ETestSuite) AddSecurityContainer(ctx context.Context) {
+	b := bytes.NewBufferString(`{  "description": "Elastic Defend Trusted Apps List",
+            "name": "Elastic Defend Trusted Apps List",
+            "list_id": "endpoint_trusted_apps",
+            "type": "endpoint",
+            "namespace_type": "agnostic"}`)
+	req, err := http.NewRequestWithContext(ctx, "POST", "http://localhost:5601/api/exception_lists", b)
+	suite.Require().NoError(err)
+	req.SetBasicAuth(suite.elasticUser, suite.elasticPass)
+	req.Header.Set("kbn-xsrf", "e2e-setup")
+	resp, err := suite.client.Do(req)
+	suite.Require().NoError(err)
+	resp.Body.Close()
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusConflict {
+		suite.Require().Failf("bad status", "status of 200 or 409 expected got %d", resp.StatusCode)
+	}
+}
+
+func (suite *BaseE2ETestSuite) AddSecurityContainerItem(ctx context.Context) string {
+	b := bytes.NewBufferString(`{
+          "description": "TEST",
+          "entries": [{
+            "field": "process.executable.caseless",
+            "value": "/bin/bash",
+            "type": "match",
+            "operator": "included"
+          }],
+          "list_id": "endpoint_trusted_apps",
+          "name": "TEST",
+          "namespace_type": "agnostic",
+          "os_types": ["linux"],
+          "type": "simple"
+        }`)
+	req, err := http.NewRequestWithContext(ctx, "POST", "http://localhost:5601/api/exception_lists/items", b)
+	suite.Require().NoError(err)
+	req.SetBasicAuth(suite.elasticUser, suite.elasticPass)
+	req.Header.Set("kbn-xsrf", "e2e-setup")
+	resp, err := suite.client.Do(req)
+	suite.Require().NoError(err)
+	defer resp.Body.Close()
+	suite.Require().Equal(http.StatusOK, resp.StatusCode)
+	var obj struct {
+		ID string `json:"id"`
+	}
+	err = json.NewDecoder(resp.Body).Decode(&obj)
+	suite.Require().NoError(err)
+	return obj.ID
+}
```

Comment:
I'm not sure if this is actually needed for the artifacts test.
Or how to reference whatever this creates in the fleet-server endpoint

Reply:
you could check with @kevinlog to clarify the use case of artifacts API